tests/test_pipelines_common.py

```diff
@@ -28,17 +28,22 @@ class PipelineTesterMixin:
     equivalence of dict and tuple outputs, etc.
     """
 
-    allowed_required_args = [
-        "source_prompt",
-        "prompt",
-        "image",
-        "mask_image",
-        "example_image",
-        "class_labels",
-        "token_indices",
-    ]
-    required_optional_params = ["generator", "num_inference_steps", "return_dict"]
-    num_inference_steps_args = ["num_inference_steps"]
+    # Canonical parameters that are passed to `__call__` regardless
+    # of the type of pipeline. They are always optional and have common
+    # sense default values.
+    required_optional_params = frozenset(
+        [
+            "num_inference_steps",
+            "num_images_per_prompt",
+            "eta",
```

Comment:
```suggestion
```

I'd remove `"eta"` here. `"eta"` is a very specific case that doesn't apply to all pipelines. We probably shouldn't have added this at all in the first place